src/Workspaces/Core/Portable/PatternMatching/SimplePatternMatcher.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Globalization;
+
+namespace Microsoft.CodeAnalysis.PatternMatching
+{
+    internal partial class PatternMatcher
+    {
+        private sealed partial class SimplePatternMatcher : PatternMatcher
```

Comment:
We're flexible on that naming pattern.  For exaple, if there are a lot of top level types, then the dot-form can be nice to make it clear that these are nested types.  When everything is nested and the entire directly is essentially one single type with a lot of nested types, we are ok just having the files simply named.